Tests/MongoSwiftTests/ReadPreferenceTests.swift

```diff
@@ -89,4 +89,99 @@ final class ReadPreferenceTests: MongoSwiftTestCase {
         expect(try ReadPreference(.secondaryPreferred, maxStalenessSeconds: 90))
             .to(equal(try ReadPreference(.secondaryPreferred, maxStalenessSeconds: 90)))
     }
+
+    func testOperationReadPreference() throws {
+        // setup a collection
+        let client = try MongoClient()
+        let db = client.db(type(of: self).testDatabase)
+        defer { try? db.drop() }
+        let coll = try db.createCollection(self.getCollectionName(suffix: "1"))
+
+        let command: Document = ["count": coll.name]
+
+        // expect runCommand to return a success response when passing in a valid read preference
+        let opts = RunCommandOptions(readPreference: ReadPreference(.secondary))
+        let res = try db.runCommand(command, options: opts)
+        expect((res["ok"] as? BSONNumber)?.doubleValue).to(bsonEqual(1.0))
+
+        // expect running other commands to not throw errors when passing in a valid read preference
+        expect(try coll.find(options: FindOptions(readPreference: ReadPreference(.primary)))).toNot(throwError())
+
+        expect(try coll.aggregate([["$project": ["a": 1] as Document]],
+                                  options: AggregateOptions(readPreference: ReadPreference(.secondary))))
+                                  .toNot(throwError())
+
+        expect(try coll.count(options: CountOptions(readPreference: ReadPreference(.secondary)))).toNot(throwError())
+
+        expect(try coll.distinct(fieldName: "a",
+                                 options: DistinctOptions(readPreference: ReadPreference(.secondary))))
+                                 .toNot(throwError())
+    }
+
+    func testClientReadPreference() throws {
+        let primaryPreferred = ReadPreference(.primaryPreferred)
+        let secondary = ReadPreference(.secondary)
+
+        do {
+            // expect that a client with an unset read preference has it default to primary
+            let client = try MongoClient()
+            expect(client.readPreference.mode).to(equal(.primary))
```

Comment:
I think it would be better to directly compare client.readPreference to a `ReadPreference(.primary)`. and similar for the other comparisons throughout this test

I know the ReadConcern and WriteConcern tests currently for some reason just read particular properties but I don't really know why as we made the types equatable. so I am fixing that in one of my pooling PRs where i added more tests for ReadConcern and WriteConcern